ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -847,20 +833,40 @@ let start () =
   in
   let configuration = Configuration.default in
   let detached = Fiber.Pool.create () in
-  let dune = Dune.create () in
+  let server = Fdecl.create Dyn.Encoder.opaque in
+  let diagnostics =
+    let workspace_root =
+      lazy
+        (let server = Fdecl.get server in
+         let state = Server.state server in
+         State.workspace_root state)
+    in
+    Diagnostics.create ~workspace_root (fun diagnostics ->
+        let server = Fdecl.get server in
+        let state = Server.state server in
+        task_if_running state ~f:(fun () ->
+            let batch = Server.Batch.create server in
+            List.iter diagnostics ~f:(fun d ->
+                Server.Batch.notification batch (PublishDiagnostics d));
+            Server.Batch.submit batch))
+  in
+  let dune = Dune.create diagnostics in
```

Comment:
If I understand, these are general diagnostics, not necessarily dune-generated diagnostics, right? Is there a reason for passing them to `Dune`?

Reply:
`Diagnostics.t` is the api for sending ocamllsp's diagnostics. Seems like dune needs to use it to send its diagnostics.